src/http.cc

```diff
@@ -2247,18 +2247,44 @@ copyOneHeaderFromClientsideRequestToUpstreamRequest(const HttpHeaderEntry *e, co
             hdr_out->addEntry(e->clone());
         break;
 
-    case Http::HdrType::IF_NONE_MATCH:
-        /** \par If-None-Match:
-         * append if the wildcard '*' special case value is present, or
+    case Http::HdrType::IF_MATCH:
+
+    case Http::HdrType::IF_NONE_MATCH: {
+        /** \par If-Match:, If-None-Match:
+         * Append if:
          *   cache_miss_revalidate is enabled, or
          *   the request is not cacheable in this proxy, or
-         *   the request contains authentication credentials.
-         * \note this header lists a set of responses for the server to elide sending. Squid added values are extending that set.
+         *   the request contains authentication credentials, or
+         *   (for If-Match) the If-None-Match header is using wildcard '*' special case value.
+         *   (for If-None-Match) the If-Match header is using wildcard '*' special case value.
+         *
+         * \note If-None-Match lists a set of responses for the server to elide sending.
+         *       Squid added values are extending that set, unless the wildcard '*' exists.
+         *
+         * \note If-Match lists a set of responses for the server to include sending.
+         *       Squid added values are extending that set, unless the wildcard '*' exists.
```

Comment:
Squid forwards (some of) the received If-Match values, but does not add its own, does it? In other words, Squid is not "extending that set". I can help rephrase this part of the comment to avoid this false implication, but it is probably best done after bigger concerns (affecting this comment) are addressed. This change request can be used to track that future update.